vendor/bat-native-ads/src/bat/ads/internal/ad_diagnostics/ad_diagnostics.cc

```diff
@@ -0,0 +1,78 @@
+/* Copyright 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/ad_diagnostics/ad_diagnostics.h"
+
+#include <string>
+#include <utility>
+
+#include "base/json/json_writer.h"
+#include "base/values.h"
+#include "bat/ads/internal/ad_diagnostics/ad_diagnostics_ads_enabled.h"
+#include "bat/ads/internal/ad_diagnostics/ad_diagnostics_catalog_id.h"
+#include "bat/ads/internal/ad_diagnostics/ad_diagnostics_catalog_last_updated.h"
+#include "bat/ads/internal/ad_diagnostics/ad_diagnostics_entry.h"
+#include "bat/ads/internal/ad_diagnostics/ad_diagnostics_last_unidle_timestamp.h"
+#include "bat/ads/internal/ad_diagnostics/ad_diagnostics_locale.h"
+#include "bat/ads/internal/ad_diagnostics/ad_diagnostics_util.h"
+
+namespace ads {
+
+namespace {
+
+AdDiagnostics* g_ad_diagnostics = nullptr;
+
+}  // namespace
+
+AdDiagnostics::AdDiagnostics() {
+  DCHECK(!g_ad_diagnostics);
+  g_ad_diagnostics = this;
+
+  SetDiagnosticsEntry(std::make_unique<AdDiagnosticsAdsEnabled>());
+  SetDiagnosticsEntry(std::make_unique<AdDiagnosticsLocale>());
+  SetDiagnosticsEntry(std::make_unique<AdDiagnosticsCatalogId>());
+  SetDiagnosticsEntry(std::make_unique<AdDiagnosticsCatalogLastUpdated>());
+  SetDiagnosticsEntry(std::make_unique<AdDiagnosticsLastUnIdleTimestamp>());
```

Comment:
Fill ad_diagnostics_entries_ map with diagnostics entries, that will be shown on internals page